m_tunnel_key.go

```diff
@@ -0,0 +1,146 @@
+package tc
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/mdlayher/netlink"
+)
+
+const (
+	tcaTunnelUnspec = iota
+	tcaTunnelKeyTm
+	tcaTunnelKeyParms
+	tcaTunnelKeyEncIPv4Src
+	tcaTunnelKeyEncIPv4Dst
+	tcaTunnelKeyEncIPv6Src
+	tcaTunnelKeyEncIPv6Dst
+	tcaTunnelKeyEncKeyID
+	tcaTunnelKeyPad
+	tcaTunnelKeyEncDstPort
+	tcaTunnelKeyNoCSUM
+	tcaTunnelKeyEncOpts
+	tcaTunnelKeyEncTOS
+	tcaTunnelKeyEncTTL
+)
+
+// TunnelKey contains attribute of the TunnelKey discipline
+type TunnelKey struct {
+	Parms           *TunnelParms
+	Tm              *Tcft
+	KeyEncIPv4Src   *net.IP
+	KeyEncIPv4Dst   *net.IP
```

Comment:
Can you change this to `KeyEncSrc` and `KeyEncDst` and check in `marshalTunnelKey` if it is IPv4 or IPv6 and set the attribute accordingly?

Reply:
Done. I just converted the net.IP object to bytes - works for both IPv6 and IPv4.